### PR TITLE
AM-CCD_Docker_Custom_Setup changes_do_not_merge

### PR DESCRIPTION
**Pull request by dharmendra8186**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/AM-589

### Change description ###
**Please don't merge this PR** .
These changes are required to support local development for case document api and other dependent applications(CCD Data Store,CCD Api Gateway, DM-Store.)
CCD Docker Application enhancement to introduce ccd case document am api  docker container configuration under ccd namespace.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: compose/backend.yml [hunk 1]
```diff
@@ -79,18 +79,43 @@ services:
...
-      # - 5005:5005
+      #- 5005:5005
     depends_on:
       - ccd-shared-database
       - ccd-user-profile-api
       - ccd-definition-store-api
     networks:
       - ccd-network
 
+  ccd-case-document-am-api:
```
Comment (mario-paniccia) [inline compose/backend.yml]: `backend.yml` is reserved for core CCD services that are started by default with `./ccd compose up`
ccd-case-document-am-api should not be here. It would be better to place is in:
`dm-store.yml`
So that it's started only when dm store is started

File: compose/backend.yml [hunk 2]
```diff
@@ -57,7 +57,7 @@ services:
       - ccd-network
 
   ccd-data-store-api:
-    image: "${CCD_DATA_STORE_API_USE_LOCAL-hmctspublic.azurecr.io/}ccd/data-store-api:${CCD_DATA_STORE_API_TAG:-latest}"
+    image: "hmctspublic.azurecr.io/ccd/data-store-api:pr-842"
```
Comment (mario-paniccia) [inline compose/backend.yml]: why this change?

File: compose/frontend.yml [hunk 3]
```diff
@@ -60,7 +60,7 @@ services:
     - ccd-network
 
   ccd-api-gateway:
-    image: "${CCD_API_GATEWAY_USE_LOCAL-hmctspublic.azurecr.io/}ccd/api-gateway-web:${CCD_API_GATEWAY_TAG:-latest}"
+    image: "hmctspublic.azurecr.io/ccd/ccd-api-gateway"
```
Comment (mario-paniccia) [inline compose/frontend.yml]: why this change?

File: compose/backend.yml (hunk 1 above)
Comment (nathan-clark) [reply compose/backend.yml]: I suggest we add to README too

File: compose/backend.yml [hunk 4]
```diff
@@ -120,7 +146,6 @@ services:
     environment:
       DB_USERNAME:
       DB_PASSWORD:
-      POSTGRES_HOST_AUTH_METHOD: trust
```
Comment (nathan-clark) [inline compose/backend.yml]: why remove?

File: compose/backend.yml (hunk 1 above)
Comment (dharmendra8186) [reply compose/backend.yml]: Yes, I have moved case document setup in dm-store.yaml fie.

Comment (dharmendra8186) [reply compose/backend.yml]: Yes, Nathan we are trying to consolidate all the changes once our testing is done read me file will be updated accordingly.

File: compose/backend.yml (hunk 2 above)
Comment (dharmendra8186) [reply compose/backend.yml]: This changes are only for our local development setup for testing and not intent to merge this PR.Sorry, for the confusion I have updated the PR description above.

File: compose/frontend.yml (hunk 3 above)
Comment (dharmendra8186) [reply compose/frontend.yml]: This changes are only for our local development setup for testing and not intent to merge this PR.Sorry, for the confusion I have updated the PR description above.

File: compose/backend.yml (hunk 4 above)
Comment (dharmendra8186) [reply compose/backend.yml]: I think, it was removed by mistake I have reverted back.

File: compose/backend.yml (hunk 2 above)
Comment (mario-paniccia) [reply compose/backend.yml]: Then why raising a PR if you don't need to merge this @dharmendra8186 ? Can you close it please

Comment (dharmendra8186) [reply compose/backend.yml]: Yes, Mario, I have sorted this PR according to the comments.Actually, ccd-case-document-am-api will be update in CCD Docker once all code will be merge into master branch of our API.However, Intention of this PR to test end to end flow with dependents micro-service based on current development.

File: compose/backend.yml (hunk 1 above)
Comment (MSancaktutar) [reply compose/backend.yml]: @dharmendra8186 , @nitishsingla1234 , @aashishaggarwal123  - Could you please confirm the above comments are actioned fully and README updated precisely as per the latest facts?

Comment (dharmendra8186) [reply compose/backend.yml]: yes, all the comments are incorporated and updated read me file is available in the PR.

File: compose/dm-store.yml [hunk 5]
```diff
@@ -68,11 +69,42 @@ services:
       - azure-storage-emulator-azurite
     ports:
       - 4506:8080
-    #   debug port exposed
+    #   debug port exposedF
```
Comment (MSancaktutar) [inline compose/dm-store.yml:72]: `exposedF` ?
